src/components/TopicBar.jsx

```diff
@@ -0,0 +1,55 @@
+import { API } from "../api";
+
+import { useState, useEffect } from "react";
+import { Link, useSearchParams } from "react-router-dom";
+
+export default function TopicBar()
+{
+    let [searchParams, setSearchParams] = useSearchParams();
+    const topic = searchParams.get("topic");
+
+    const [topics, setTopics] = useState([]);
+
+    useEffect(() => {
+        const fetchTopics = async () => {
+            try
+            {
+                const response = await API.get("/topics");
+                setTopics(response.data.topics);
```

Comment:
Great that you're using an API request to get the list of topics rather than hard coding it 👍